### DISCUSSION TITLE
Config option to stop server if plugins failed to load

### DISCUSSION THREAD

**Discussion by Astralchroma** [PaperMC/Paper#10562] [Ideas]
### Is your feature request related to a problem?

If a critical plugin fails to load, such as a claim protection plugin, but the server continues to start anyway and allows players to join this would put the server in a state where players are able to do things they otherwise wouldn't

### Describe the solution you'd like.

A "required-plugins" config option which has a list of plugin names which the server will stop in the event they fail to load
Additionally it would be nice to just specify "*" to require that all plugins load

### Describe alternatives you've considered.

I have attempted to put together a PR for this but I have ran into issues with the config not yet being available during paper plugin bootstrapping, if someone wants to help with this, that would be great

**Comment by Xander-de-Keijzer** [comment]
Is this only during startup, what in the case that a plugin is unloaded by plugman for example, or doesn't load after a reload?

**Comment by Astralchroma** [comment]
both reloading and plugman are already unsupported so it doesnt matter
regardless the server would still stop if the plugins fail to load regardless of why that plugin was loaded

**Comment by Brokkonaut** [comment]
I would prefer to have this for all plugins by default, maybe with some startup option as opt-out or so

**Comment by SlimeDog** [comment]
We manage this with [RequiredPlugins](https://www.spigotmc.org/resources/67281/). Players cannot connect if the required plugins are not enabled.

**Comment by Machine-Maker** [comment]
Is there a reason why a plugin can't just handle this themselves? Listen to the load event and checking if all plugins are enabled. It's such a simple thing for a plugin to handle, I don't know if its worth adding it.

**Comment by electronicboy** [comment]
and if that plugin somehow happen chance blows up?
afaik, this is something that aikar has in his emc fork that he wanted to port over given that it's useful for ensuring that important stuff loads; while yea, you could have a plugin that implements this, it's not 100% fool proof

**Comment by SlimeDog** [comment]
> and if that plugin somehow happen chance blows up?

We have recently replaced RequirePlugins with [PluginControl](https://www.spigotmc.org/resources/plugincontrol.110590/). We monitor server restart, to ensure that PluginControl doesn't "blow up." If one allows auto-restart, it is straightforward to script to check that it starts with whatever logic is desired.

**Comment by notTamion** [comment]
just going to bump this and ask people if this is still something you want? there are some fair points against having this in paper (e.g. plugins not working correctly even though they are enabled and that giving a sense of false security)

links to the discord convos about it:
https://discord.com/channels/289587909051416579/1317066157727023156/1320790838393180222
https://discord.com/channels/289587909051416579/925530366192779286/1325094033122394134
https://discord.com/channels/289587909051416579/925530366192779286/1180689570694697043

**Comment by Brokkonaut** [comment]
Yes I definitely think this would be useful. yes this gives no 100% security (nothing can give that), but it can avoid accidential starting the server when there definitely is a problematic misconfiguration
